app/models/activity.rb

```diff
@@ -327,4 +327,8 @@ def requires_policy_markers?
   def is_project?
     project? || third_party_project?
   end
+
+  def is_gcrf_funded?
+    associated_fund&.roda_identifier_fragment == "GCRF"
```

Comment:
Is there a situation where `associated_fund` returns `nil`? When does that happen?

Reply:
Yes, this was deliberate - see https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/pull/760/files/2ac93b03a09e738c9120f9c0fa868cf7b2e4aff4#diff-cc9fe353a052960d819ee781d541ddc49305a1982a64712769f1d8a17b96ed6fR1210-R1214

When an activity is first created it is missing basic attributes which determine which `level` it is and what `parent` it belongs to. It makes sense for `associated_fund` to be `nil` in these cases, which is why we need the safe navigation operator here.

Reply:
Of course, I could be more explicit, i.e.

```ruby
def is_gcrf_funded?
  return false if associated_fund.nil?

  associated_fund.roda_identifier_fragment == "GCRF"
end
```